digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java

```diff
@@ -1595,22 +1643,50 @@ StoredSession upsertAndLockSession(@Bind("projectId") int projectId,
 
         List<StoredSessionWithLastAttempt> getSessions(@Bind("siteId") int siteId, @Bind("limit") int limit, @Bind("lastId") long lastId);
 
+        List<StoredSessionWithLastAttempt> getSessions(@Bind("siteId") int siteId, @Bind("limit") int limit, @Bind("lastId") long lastId, @Bind("offset") int offset);
+
+        @SqlQuery("select count(1)" +
+                " from sessions s" +
+                " join session_attempts sa on sa.id = s.last_attempt_id" +
+                " where s.project_id in (select id from projects where site_id = :siteId)" +
+                " and s.id < :lastId")
+        Integer getSessionsCount(@Bind("siteId") int siteId, @Bind("lastId") long lastId);
+
+
+        @SqlQuery("select count(1)" +
+                " from sessions s" +
+                " join session_attempts sa on sa.id = s.last_attempt_id" +
+                " where s.project_id = :projectId" +
+                " and s.id < :lastId")
+        Integer getSessionsCountOfProject(@Bind("lastId") long lastId, @Bind("projectId") int projectId);
```

Comment:
It looks like this query is different from getSessionsOfProject. Not have site_id condition.
Could you fix it and add test code for *Count() method.
In addition the test data should include different site_id and check.